src/VisualStudio/VisualBasic/Impl/LanguageService/VisualBasicPackage.vb

```diff
@@ -84,40 +84,43 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic
             Return Me.ComponentModel.GetService(Of VisualStudioWorkspaceImpl)
         End Function
 
-        Protected Overrides Sub Initialize()
+        Protected Overrides Async Function InitializeAsync(cancellationToken As CancellationToken, progress As IProgress(Of ServiceProgressData)) As Task
             Try
-                MyBase.Initialize()
+                Await MyBase.InitializeAsync(cancellationToken, progress).ConfigureAwait(True)
+                Await JoinableTaskFactory.SwitchToMainThreadAsync()
 
                 RegisterLanguageService(GetType(IVbCompilerService), Function() _comAggregate)
 
                 Dim workspace = Me.ComponentModel.GetService(Of VisualStudioWorkspaceImpl)()
                 RegisterService(Of IVbTempPECompilerFactory)(Function() New TempPECompilerFactory(workspace))
 
-                RegisterObjectBrowserLibraryManager()
+                Await RegisterObjectBrowserLibraryManagerAsync().ConfigureAwait(True)
             Catch ex As Exception When FatalError.Report(ex)
             End Try
-        End Sub
+        End Function
 
         Protected Overrides Sub Dispose(disposing As Boolean)
-            UnregisterObjectBrowserLibraryManager()
+            If disposing Then
+                JoinableTaskFactory.Run(AddressOf UnregisterObjectBrowserLibraryManagerAsync)
+            End If
 
             MyBase.Dispose(disposing)
         End Sub
 
-        Private Sub RegisterObjectBrowserLibraryManager()
-            Dim objectManager = TryCast(Me.GetService(GetType(SVsObjectManager)), IVsObjectManager2)
+        Private Async Function RegisterObjectBrowserLibraryManagerAsync() As Task
+            Dim objectManager = TryCast(Await GetServiceAsync(GetType(SVsObjectManager)).ConfigureAwait(True), IVsObjectManager2)
```

Comment:
Same as in C#: should you switch to the main thread explicitly before casting `SVsObjectManager`?